hutch_python/epics_arch.py

```diff
@@ -95,11 +140,204 @@ def create_arch_file(experiment, hutch=None, path=None, dry_run=False):
+    """
+    Gather all user objects from the CDS tab in the questionnaire.
+    Parse objects and separate them based on type.
+    Display them in the console via PrettyTable.
+
+    Parameters
+    ----------
+    exp: ``str``
+        The experiment's name e.g. xppx1003221
+    run: ''str''
+        The run number e.g. run21
+
+    Outputs
+    -------
+        PrettyTable visualization of cds objects
+
+
+    """
+    """
+    pull run data from questionnaire api, then take the data and sort it
+    create Pretty Table instance and if the values from the run data contain pcdssetup
+    then put them into a seperate dictionary as they are cds items
+    """
+    logger.debug('pull_cds_items(%s)', exp)
+    client = QuestionnaireClient()
+    logger.debug("in cds items, run numb:", str(run[1]))
+    runDetails_Dict = client.getProposalDetailsForRun(str(run[0]), str(run[1]))
+    sorted_runDetails_Dict = dict(sorted(runDetails_Dict.items()))
+    cds_dict = {}
+    myTable = PrettyTable(["Alias", "PV Base", "Type"])
+    for keys, vals in sorted_runDetails_Dict.items():
+        if "pcdssetup" in keys:
+            cds_dict[keys] = vals
+
+    """
+    names are as follows:
+    pcdssetup-motors, pcdssetup-areadet, pcdssetup-ao, pcdssetup-devs
+    pcdssetup-ps, pcdssetup-trig, pcdssetup-vacuum, pcdssetup-temp
+
+    iterate through all cds items and label them based on their type
+    use the struct members to identify
+    """
+    displayList = []
+    for k, v in cds_dict.items():
+        if re.match('pcdssetup-motors.*-name', k):
+            pv = cds_dict.get(re.sub('name', 'pvbase', k), '')
+            displayList.append(QStruct(v, pv, "motors"))
+        elif re.match('pcdssetup-areadet.*-name', k):
+            pv = cds_dict.get(re.sub('name', 'pvbase', k), '')
+            displayList.append(QStruct(v, pv, "areadet"))
+        elif re.match('pcdssetup-ao.*-name', k):
+            pv = cds_dict.get(re.sub('name', 'pvbase', k), '')
+            displayList.append(QStruct(v, pv, "analog output"))
+        elif re.match('pcdssetup-devs.*-name', k):
+            pv = cds_dict.get(re.sub('name', 'pvbase', k), '')
+            displayList.append(QStruct(v, pv, "other devices"))
+        elif re.match('pcdssetup-ps.*-name', k):
+            pv = cds_dict.get(re.sub('name', 'pvname', k), '')
+            displayList.append(QStruct(v, pv, "power supplies"))
+        elif re.match('pcdssetup-trig.*-name', k):
+            pv = cds_dict.get(re.sub('name', 'pvbase', k), '')
+            displayList.append(QStruct(v, pv, "triggers"))
+        elif re.match('pcdssetup-vacuum.*-name', k):
+            pv = cds_dict.get(re.sub('name', 'pvbase', k), '')
+            displayList.append(QStruct(v, pv, "vacuum"))
+        elif re.match('pcdssetup-temp.*-name', k):
+            pv = cds_dict.get(re.sub('name', 'pvbase', k), '')
+            displayList.append(QStruct(v, pv, "temperature"))
+    # logger.debug("displayList", displayList)
```

Comment:
nitpick: hanging debug comment, imo you can either uncomment it and leave it in or delete (or ignore the nitpick)